static/app/components/events/featureFlags/setupIntegrationModal.tsx

```diff
@@ -0,0 +1,216 @@
+import {Fragment, useCallback, useState} from 'react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import {addSuccessMessage} from 'sentry/actionCreators/indicator';
+import type {ModalRenderProps} from 'sentry/actionCreators/modal';
+import Alert from 'sentry/components/alert';
+import {Button, LinkButton} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import type {Data} from 'sentry/components/forms/types';
+import TextCopyInput from 'sentry/components/textCopyInput';
+import {IconWarning} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import OrganizationStore from 'sentry/stores/organizationStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import {space} from 'sentry/styles/space';
+import {defined} from 'sentry/utils';
+import useApi from 'sentry/utils/useApi';
+
+export type ChildrenProps<T> = {
+  Body: (props: {children: React.ReactNode}) => ReturnType<ModalRenderProps['Body']>;
+  Footer: () => ReturnType<ModalRenderProps['Footer']>;
+  Header: (props: {children: React.ReactNode}) => ReturnType<ModalRenderProps['Header']>;
+  state: T;
+};
+
+interface State {
+  provider: string;
+  url: string | undefined;
+}
+
+function useGenerateAuthToken(state: State) {
+  const api = useApi();
+  const date = new Date().toISOString();
+
+  const createToken = async () =>
+    await api.requestPromise(`/api-tokens/`, {
+      method: 'POST',
+      data: {
+        name: `${state.provider} Token ${date}`,
+        scopes: [
+          'project:read',
+          'project:write',
+          'project:admin',
+          'event:read',
+          'event:write',
+          'event:admin',
+          'org:read',
+          'org:write',
+          'org:admin',
```

Comment:
assuming these are the only scopes we need

Reply:
Let's use this endpoint `https://sentry.io/api/0/organizations/colton-test/org-auth-tokens/` with this payload `{"name": token_name}`.

Reply:
Scope will be set automatically as org:ci.

Reply:
cool! looks like it worked

<img width="965" alt="SCR-20241108-jaqa" src="https://github.com/user-attachments/assets/7e272806-e8b6-456e-ac0b-083a3bc7362d">
<img width="691" alt="SCR-20241108-jans" src="https://github.com/user-attachments/assets/1a220a78-ebd1-43a0-8cbd-cd6771fc1fec">